# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: parser/html/nsHtml5TokenizerHSupplement.h
# Commit: 3dfd3c94a105
# Full Hash: 3dfd3c94a105e095aada0b356f1106370de222d3
# Author: Henri Sivonen <hsivonen@hsivonen.fi>
# Date: 2021-12-07 15:51:17
# Regressor Bug: 1701828
# File Overlap Count: 1
# Description:
#   Bug 1701828 - meta charset rewrite. r=smaug
#   
#   Implements https://github.com/whatwg/html/issues/6962 . Improves performance
#   when <meta charset> occurs in head but after the first kilobyte and aligns
#   behavior better with WebKit and Blink.
# ==============================================================================

diff --git a/parser/html/nsHtml5TokenizerHSupplement.h b/parser/html/nsHtml5TokenizerHSupplement.h
--- a/parser/html/nsHtml5TokenizerHSupplement.h
+++ b/parser/html/nsHtml5TokenizerHSupplement.h
@@ -9,32 +9,48 @@ inline nsHtml5HtmlAttributes* GetAttribu
  * UTF-16 code units before having to make the buffers larger.
  *
  * @param aLength the number of UTF-16 code units to be tokenized before the
  *                next call to this method.
  * @return true if successful; false if out of memory
  */
 bool EnsureBufferSpace(int32_t aLength);
 
+bool TemplatePushedOrHeadPopped();
+
+void RememberGt(int32_t aPos);
+
+void AtKilobyteBoundary() { suspendAfterCurrentTokenIfNotInText(); }
+
+bool IsInTokenStartedAtKilobyteBoundary() {
+  return suspensionAfterCurrentNonTextTokenPending();
+}
+
 mozilla::UniquePtr<nsHtml5Highlighter> mViewSource;
 
 /**
  * Starts handling text/plain. This is a one-way initialization. There is
  * no corresponding EndPlainText() call.
  */
 void StartPlainText();
 
 void EnableViewSource(nsHtml5Highlighter* aHighlighter);
 
 bool FlushViewSource();
 
 void StartViewSource(const nsAutoString& aTitle);
 
+void StartViewSourceCharacters();
+
 void EndViewSource();
 
+void RewindViewSource();
+
+void SetViewSourceOpSink(nsAHtml5TreeOpSink* aOpSink);
+
 void errGarbageAfterLtSlash();
 
 void errLtSlashGt();
 
 void errWarnLtSlashInRcdata();
 
 void errCharRefLacksSemicolon();
 